From e09e752949e7af0231884d1b11ea907e3e8b1611 Mon Sep 17 00:00:00 2001
From: Artem Gindinson <artem.gindinson@intel.com>
Date: Fri, 18 Nov 2022 14:48:45 +0000
Subject: [PATCH]  Additional VC fix of pointer element type API usage

Quick fixup to the LLVM 15 buildability in the footsteps of commit
intel/intel-graphics-compiler@b86c1bf.
---
 IGC/VectorCompiler/lib/GenXCodeGen/GenXSimdCFRegion.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/IGC/VectorCompiler/lib/GenXCodeGen/GenXSimdCFRegion.cpp b/IGC/VectorCompiler/lib/GenXCodeGen/GenXSimdCFRegion.cpp
index 363d622c24c..c047c545ac8 100644
--- a/IGC/VectorCompiler/lib/GenXCodeGen/GenXSimdCFRegion.cpp
+++ b/IGC/VectorCompiler/lib/GenXCodeGen/GenXSimdCFRegion.cpp
@@ -748,12 +748,12 @@ void GenXPredToSimdCF::insertIfGoto(SimdCFIfRegion &R) {
                                     Mask->getName() + ".not");
   Value *EM = getEM(M);
   Instruction *OldGotoEM =
-      Builder.CreateLoad(EM->getType()->getPointerElementType(), EM,
+      Builder.CreateLoad(IGCLLVM::getNonOpaquePtrEltTy(EM->getType()), EM,
                          false /*isVolatile*/, EM->getName());
   OldGotoEM->setDebugLoc(DL);
   Value *GotoRMAddr = getRMAddr(Br->getSuccessor(0), SimdWidth);
   Instruction *OldGotoRM = Builder.CreateLoad(
-      GotoRMAddr->getType()->getPointerElementType(), GotoRMAddr,
+      IGCLLVM::getNonOpaquePtrEltTy(GotoRMAddr->getType()), GotoRMAddr,
       false /*isVolatile*/, GotoRMAddr->getName());
   OldGotoRM->setDebugLoc(DL);
   Type *GotoTys[] = {OldGotoEM->getType(), OldGotoRM->getType()};
@@ -798,12 +798,12 @@ void GenXPredToSimdCF::insertIfGoto(SimdCFIfRegion &R) {
   AfterThens[&R] = AfterThen;
 
   Builder.SetInsertPoint(AfterThen, AfterThen->begin());
-  auto *OldEM = Builder.CreateLoad(EM->getType()->getPointerElementType(), EM,
+  auto *OldEM = Builder.CreateLoad(IGCLLVM::getNonOpaquePtrEltTy(EM->getType()), EM,
                                    false /*isVolatile*/, EM->getName());
   OldEM->setDebugLoc(DL);
   auto *RMAddr = getRMAddr(Br->getSuccessor(0), SimdWidth);
   auto *RM =
-      Builder.CreateLoad(RMAddr->getType()->getPointerElementType(), RMAddr,
+      Builder.CreateLoad(IGCLLVM::getNonOpaquePtrEltTy(RMAddr->getType()), RMAddr,
                          false /*isVolatile*/, RMAddr->getName());
   RM->setDebugLoc(DL);
   Type *Tys[] = {OldEM->getType(), RM->getType()};
